app/src/features/surveys/observations/observations-table/observation-row-validation/ObservationRowValidationUtils.ts

```diff
@@ -0,0 +1,236 @@
+      );
+      if (foundQuantitative) {
+        const error = _validateQuantitativeMeasurement(
+          measurementColumn,
+          Number(data),
+          foundQuantitative.min_value,
+          foundQuantitative.max_value
+        );
+
+        if (error) {
+          measurementErrors.push(error);
+        }
+      }
+
+      // A measurement column has data but no measurements were found for the itis_tsn
+      if (!foundQualitative && !foundQuantitative) {
+        measurementErrors.push({
+          field: measurementColumn,
+          message: `Invalid measurement set for taxon.`
+        });
+      }
+    }
+  });
+
+  return measurementErrors;
+};
+
+/**
+ * This validates if the provided option UUID exists within the given list of CBQualitativeOption.
+ * Returns null if the option is valid or an ObservationRowValidationError if it is not found
+ *
+ * @param {string} field The column key for the data being validated
+ * @param {string} value The options UUID to look for
+ * @param {CBQualitativeOption[]} options
+ * @returns {*} ObservationRowValidationError | null
+ */
+const _validateQualitativeMeasurement = (
+  field: string,
+  value: string,
+  options: CBQualitativeOption[]
+): ObservationRowValidationError | null => {
+  const foundOption = options.find((op) => op.qualitative_option_id === value);
+
+  if (!foundOption) {
+    // found measurement, no option, no bueno
+    return {
+      field,
+      message: 'Invalid option selected for taxon.'
+    };
+  }
+  return null;
+};
+
+/**
+ *  Validates if a given value is in between min and max values from a Quantitative Measurement definition in Critterbase.
+ *  Returns null if the value is valid or a ObservationRowValidationError describing that the value is out of the valid range.
+ *
+ * @param {string} field The column key for the data being validated
+ * @param {number} value The number to validate
+ * @param {number | null} minValue The min value provided by the measurement definition from Critterbase
+ * @param {number | null} maxValue The max value provided by the measurement definition from Critterbase
+ * @returns {*} ObservationRowValidationError | null
+ */
+const _validateQuantitativeMeasurement = (
+  field: string,
+  value: number,
+  minValue: number | null,
+  maxValue: number | null
+): ObservationRowValidationError | null => {
+  if (minValue && maxValue) {
+    if (minValue <= value && value <= maxValue) {
+      return null;
+    }
+  } else {
+    if (minValue !== null && minValue <= value) {
+      return null;
+    }
+
+    if (maxValue !== null && value <= maxValue) {
+      return null;
+    }
+
+    if (minValue === null && maxValue === null) {
+      return null;
+    }
+  }
+
+  // Measurement values are invalid, create an error and return
+  return {
+    field,
+    message: `Value provided is outside of the valid range ${minValue} < ${value} < ${maxValue}`
```

Comment:
Technically the `minValue` and `maxValue` are allowed, so the range is represented by `${minValue} ≤ ${value} ≤ ${maxValue}`

Reply:
(Opinion) I probably wouldn't use an inequality to represent the range either way, since if this validation error occurs, the value  of `${value}` wouldn't be within the range, so the inequality wouldn't hold. I would just use [square brackets](https://en.wikipedia.org/wiki/Interval_(mathematics))
```javascript
message: `Value provided is outside of the valid range [${minValue}, ${maxValue}]`
```

Reply:
yeah good point, updated